### PR TITLE
#73 added project-details component

### LINKED ISSUE #73: Projects - Create Project Details Component
Create the view-only component for displaying project details. This should cleanly and readably display a set of key value pairs supplied by the projects-single API endpoint. Refer to the design sketch v4 snippet below for inspiration and context.

---
Here's a quick outline of steps for this issue. Check them off and / or suggest edits to them as you make progress on the issue. Existing dummy components may be useful to consult for guidance ([see here](https://github.com/Northeastern-Electric-Racing/PM-Dashboard-v2/blob/main/src/components/project-details/project-details.tsx)).
- [x] Create the component folder in `/src/components/`
- [x] Create the test file, component file, and ccs module in the component folder
- [x] Add basic TS functional component boilerplate to component file
- [x] Add basic test suite `describe()` to test file
- [x] Add the component to the main app component (so you can see the component when starting the app in development)
- [x] Configure component props to take in the proper work package details
- [x] Display all desired component props in the tsx
- [x] Add DOM element division to begin arranging / styling data elements
- [x] Add styling to CSS module file
- [x] Attach CSS styling to proper sections / DOM elements in tsx
- [x] Get the component to an aesthetically pleasing state that resembles the design sketch snippet below
- [x] Open a PR with a screenshot of the component to get review and feedback
- [ ] Iterate and make changes based on feedback
- [ ] Once approved for merging, attach a screenshot of the finished component to this issue

---
Design Sketch v4 Snippet:
Screen Shot 2021-02-07 at 12 03 53 AM

### PR DESCRIPTION
**Pull request by TheElbowTurnMaster**
closed #73 

My understanding of CSS is rudimentary, so this was as far as I worked on it before requesting feedback. There is a slight bug in which the WBS number is swapped for 1.2.0 and 1.12.0. At a glance, I'm not sure what the issue is.

image

### CODE REVIEW

File: src/components/project-details/project-details.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.tsx:15]: Any reason why `id` instead of `className`?

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Wow that was fast! It looks awesome too! Great work so far.

- [x] Fix linting errors (ignore the href ones for now)
- [x] Add tests
- [x] Split styling for ID into ID for unique styling and class for repeatable styling

I think we'll have to wait until my massive PR gets merged before we can effectively finish this issue with proper mock data.

File: src/components/project-details/project-details.tsx
Comment (TheElbowTurnMaster) [reply src/components/project-details/project-details.tsx:15]: I read that id was used for unique elements. Is it better to use class in this case?

Comment (jamescd18) [reply src/components/project-details/project-details.tsx:15]: Would you be able to send me the link where you read that? I'd love to learn more about why. If not, that's okay, just let me know.

Comment (TheElbowTurnMaster) [reply src/components/project-details/project-details.tsx:15]: https://css-tricks.com/the-difference-between-id-and-class/

Comment (jamescd18) [reply src/components/project-details/project-details.tsx:15]: Thank you! I actually didn't realize URLs with `...#some-tag` were done via HTML element IDs. Learn something new every day.

In this case, it's a little nuanced. You should use both an ID and a class. Use the ID for styling that is unique to the element and so that we can uniquely identify it on the page. Use the class for any styling that might end up being shared between other tiles on the page that you can see in the design sketch snippet that is attached to the issue #73.

File: src/components/project-details/project-details.module.css
Comment (jamescd18) [inline src/components/project-details/project-details.module.css:6]: With the naming of the ID, does it impact the url you need to navigate to? 

I would prefer us to use kebab case for naming IDs:
This `<url>#project-details`
Instead of this `<url>#project_details`

File: src/components/project-details/project-details.module.css
Comment (jamescd18) [inline src/components/project-details/project-details.module.css:20]: Let's move this styling definition into the top-level index.css file so that it can be used in all areas of the application.

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Looking good! I like the changes you've made so far, just a couple more small changes and then we'll wait for my PR to get merged.

File: src/components/project-details/project-details.test.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.test.tsx:20]: let's use the `it()` syntax just to keep things consistent. Behind the scenes, they're aliases so they point to the same function.

**Review by jamescd18 — CHANGES_REQUESTED** [review]
minor change, then wait for the PR

**Comment by jamescd18** [comment]
Now that the massive PR has been merged, make sure to update and fast forward your branch. Then examine the data types and new location for dummy data, and refactor / update / improve the code for this PR so it integrates nicely. Let me know if you run into any issues!

**Comment by TheElbowTurnMaster** [comment]
Since the project-table uses DisplayProject instead of Project, is it alright if I add a project field to the interface so I can send that to the project-details component?

**Comment by TheElbowTurnMaster** [comment]
Dammit I recognized the URL but didn't react in time

**Review by kevinyu328 — CHANGES_REQUESTED** [review]
Amazing work Jeff! Thank you for working on this! Just a couple super small things and then I think we're good. Also, do you have Prettier set up on your vscode?

File: src/components/project-details/project-details.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.tsx:6]:
```suggestion
import { Project } from 'utils';
```
Make sure to always be importing from the package and not the file itself.

File: src/containers/projects-table/projects-table.tsx
Comment (jamescd18) [inline src/containers/projects-table/projects-table.tsx:94]: Since we removed the state field, let's put it back the way it was before so as to not introduce unnecessary changes

**Review by jamescd18 — CHANGES_REQUESTED** [review]

File: src/components/project-details/project-details.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.tsx:33]: Why an HTML menu tag? Quick Google search says that it has very little browser support so we likely don't want to be using it.

Screen Shot 2021-04-28 at 2 49 42 PM

File: src/components/project-details/project-details.module.css
Comment (jamescd18) [inline src/components/project-details/project-details.module.css:34]: prefer using class names instead of tag references for styling

File: src/pages/wbs-details/wbs-details.tsx
Comment (jamescd18) [inline src/pages/wbs-details/wbs-details.tsx:31]: this GitHub issue is supposed to just focus on creating the view-only component and not the data fetching aspect because that will be implemented later once all the child components are ready. If you're looking to just make the view-only component available for you to view when running the application, you can just place it hard-coded into the project on a page and provide it one of the dummy data example objects.

if you're interested in the data fetching component, you should take a look at how it is done in other components. we want the design pattern to match as much as possible unless there is a good reason for it to be different. if you have suggestions on how to improve the design pattern for fetching data, I'd be happy to discuss it. see the container components for examples of data fetching.

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Let's focus this PR on just the view-only component so as to bias towards changing less of the codebase since fetching data for this component will be implemented in a different issue. Also a note on the HTML in the component, see the review comments below.

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Great job making the changes! I've added some more review comments from some additional things I noticed. Let me know if you have any questions!

File: src/components/project-details/project-details.test.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.test.tsx:21]:
```suggestion
    const projectElement = screen.getByText(wbsPipe(exampleProject2.wbsNum));
```
use the wbsPipe for this instead, and make sure to import the wbsPipe at the top too.

File: src/components/project-details/project-details.test.tsx
Comment (jamescd18) [inline src/components/project-details/project-details.test.tsx:28]: use the fullName pipe here instead of a template string

**Review by jamescd18 — CHANGES_REQUESTED** [review]
Looking better! Just two more locations to use pipes instead of template strings.

**Review by jamescd18 — APPROVED** [review]

**Pull request merged**
